### ISSUE TITLE
FirebaseAnalyticsSwift-Beta: Failure to Build/Archive with Xcode

### ISSUE THREAD

**Issue by cpdunphy** [firebase/firebase-ios-sdk#7785]
### Describe your environment

  * Xcode version: 12.4
  * Firebase SDK version: 7.9.1
  * Installation method: `Swift Package Manager`
  * Firebase Component: `FirebaseAnalyticsSwift-Beta`

### Describe the problem

Xcode Archive fails when using the `FirebaseAnalyticsSwift-Beta` component. Building and installing the app with the dependency works so far as expected, the trouble is with archiving.

Note: the archive will succeed if you copy/add the source file (`Analytics+SwiftUI.swift`) to your local project, remove the troubled component (`FirebaseAnalyticsSwift-Beta`), and then attempt an archive. Due to this, I am led to believe this is a configuration issue, not (another) issue with SwiftUI. 

Additionally, as noted by the Firebase SPM documentation, there is a preexisting issue (#6472) that also deals with analytics archive errors. Since the issue I'm detailing seems to be contained to the `FirebaseAnalyticsSwift-Beta` component, I do not currently have reason to believe the other is relevant. 

#### Steps to reproduce:

1. Create a new project
2. Add "https://github.com/firebase/firebase-ios-sdk.git" as a Package Dependency (V 7.9.1)
3. Select the desired components: `FirebaseAnalyticsSwift-Beta` and `FirebaseAnalytics`
4. Archive

#### Relevant Error Messages:
```
FirebaseAnalyticsSwift Group
Swift Compiler Error Group
- Analytics+SwiftUI.swift:35:41: Cannot find type 'View' in scope
- Analytics+SwiftUI.swift:35:24: Cannot find type 'Content' in scope
- Analytics+SwiftUI.swift:25:44: Cannot find type 'ViewModifier' in scope
- Analytics+SwiftUI.swift:53:20: Cannot find type 'View' in scope
Uncategorized Group
- Command CompileSwift failed with a nonzero exit code
```

I'd like to note this additional warning. It does not (always) seem to impede the completion of a build, but does also popup along side the error above and seems contained to the `FirebaseAnalyticsSwift-Beta` component.
```
Skipping duplicate build file in Copy Files build phase: /Users/{user}/Library/Developer/Xcode/DerivedData/{project}/SourcePackages/artifacts/Firebase/FirebaseAnalytics.xcframework/ios-arm64_armv7/FirebaseAnalytics.framework
Skipping duplicate build file in Copy Files build phase: /Users/{user}/Library/Developer/Xcode/DerivedData/{project}/SourcePackages/artifacts/GoogleAppMeasurement/GoogleAppMeasurement.xcframework/ios-arm64_armv7/GoogleAppMeasurement.framework
```

**Comment by ryanwilson** [comment]
Thanks for the report and sorry for the troubles! 

The workaround of copying the file manually sounds like a good approach for now while we figure out what's going on. I'm not sure why that's an issue though off the top of my head, as we're conditionally checking if `SwiftUI` can be imported here:

https://github.com/firebase/firebase-ios-sdk/blob/881518c82aa5882bc9fff17b60881edef1ccea9e/FirebaseAnalyticsSwift/Sources/Analytics%2BSwiftUI.swift#L15

Thanks again, we'll try to reproduce and report back with progress.

**Comment by ryanwilson** [comment]
Haven't tested this yet but I think adding a `linkedFramework` may be the solution here, similar to this inclusion of StoreKit:  https://github.com/firebase/firebase-ios-sdk/blob/881518c82aa5882bc9fff17b60881edef1ccea9e/Package.swift#L258

**Comment by cpdunphy** [comment]
@ryanwilson Thanks for taking a look! Obviously the component is in beta and issues are a given, but I figured it is worth mentioning. Let me know if you're able to reproduce it.

**Comment by ryanwilson** [comment]
> @ryanwilson Thanks for taking a look! Obviously the component is in beta and issues are a given, but I figured it is worth mentioning. Let me know if you're able to reproduce it.

Thanks, I can definitely reproduce it but the fix isn't as simple as I had originally hoped. Even with adding it as an explicitly linked framework, I'm still seeing the issue. I temporarily removed the `#if canImport(SwiftUI` and still running into the issue, despite importing SwiftUI on line 17.

I'll keep digging, thanks again for the report!

**Comment by ryanwilson** [comment]
Hmm, I suspect this is the issue: 
https://github.com/firebase/firebase-ios-sdk/blob/9f36662ee8b71dd6b6e15e5b57ce30daefafb832/Package.swift#L24

If I change that to `.iOS(.v13)` I'm able to successfully archive with a few local changes. That's a no-go in general though since the rest of the SDKs still need to support older iOS versions. I suspect it could be a bug on Apple's side, since everything should be marked appropriately and we're gating the import but maybe this is working as intended. I'll try to assemble a minimal repro for now and update the thread on any results.

In the meantime, the workaround of adding the file to your app manually is a good approach.

Re: warnings of duplicate build files, when this eventually works you can remove the `FirebaseAnalytics` target as it gets included automatically when you depend on `FirebaseAnalyticsSwift-Beta`, although that's not abundantly clear and only due to `FirebaseAnalytics` being a binary dependency.

**Comment by paulb777** [comment]
Besides archiving this is also a problem for building FirebaseAnallyticsSwift-Beta for Any iOS Device.

**Comment by Panajev** [comment]
I had to revert to Firebase 7.10.0 from 7.11.0 as it would not build with the Xcode 12.5 RC build (SwiftPM integration).

Tried to fully remove Firebase and then re-add it via SwiftPM and could only re-add 7.10.0, Xcode would throw up an error and refuse to add 7.11.0.  I think this is pretty important as Xcode 12.5 will go live very shortly as iOS 14.5 goes live this week.

**Comment by ryanwilson** [comment]
@Panajev thanks for the additional report - looking into this now.

Was the error you're seeing similar to the screenshot below? If so, that's a different error but needs attention. I've filed #7962 to track this.

![Installation error in Xcode](https://user-images.githubusercontent.com/1097316/116090372-c9acea80-a671-11eb-9a12-d6eef6bf347e.png)

**Comment by ryanwilson** [comment]
TODO for solving the original issue: create a minimal repro to report to the Swift forums / Apple's Feedback system, or ideally find a solution based on the minimal repro.

**Comment by Panajev** [comment]
Hey Ryan, I think so. I will create a sample project and reproduce it there too :)! Thanks for looking into this.

**Comment by ryanwilson** [comment]
Going to close this now that it's merged into `master` - this will be fixed for Firebase 8.0.

Thanks for the report!
